dev/docs.go

```diff
@@ -74,11 +89,40 @@ func (d Docs) Lint(ctx context.Context) error {
 		return err
 	})
 
-	eg.Go(func() error {
-		return util.DiffDirectoryF(ctx, d.Dagger.Source(), d.Generate, generatedSchemaPath, generatedCliZenPath)
+	eg.Go(func() (rerr error) {
+		ctx, span := Tracer().Start(ctx, "check that generated GraphQL reference documentation is up-to-date")
+		defer func() {
+			if rerr != nil {
+				span.SetStatus(codes.Error, rerr.Error())
+			}
+			span.End()
+		}()
+		return util.DiffDirectoryF(ctx, d.Dagger.Source(), d.Generate, generatedSchemaPath)
```

Comment:
Could be worth pushing the span setup into `util.DiffDirectoryF` by giving it a required arg for the name of the thing being diffed, like `"GraphQL reference documentation"`, and using that to construct the span name + a descriptive error message.

Reply:
I have a separate PR that removes `DiffDirectoryF` altogether in favor of `dirdiff` module. #7877 

I will look into this idea for the successor. But the span would be less expressive.